src/custom-blocks/auto-item-list/index.js

```diff
@@ -0,0 +1,468 @@
+import { store as coreStore } from '@wordpress/core-data';
+
+const { __ } = wp.i18n;
+const { registerBlockType, registerBlockStyle } = wp.blocks;
+const { Fragment } = wp.element;
+const { useSelect } = wp.data;
+const { InspectorControls, MediaUpload, InnerBlocks } = wp.blockEditor;
+const { PanelBody, SelectControl, Button, TextControl, ToggleControl, RadioControl} = wp.components;
+const allowedMediaTypes = ['image'];
+
+import { __experimentalNumberControl as NumberControl } from '@wordpress/components';
+import { __experimentalText as Text } from '@wordpress/components';
+
+registerBlockType("mojblocks/auto-item-list", {
+  title: __("Most Recent Items", "mojblocks"),
+  description: __('Auto-populates with most recent or next items based on a date'),
+  category: "mojblocks",
+  icon: "slides",
+  keywords: [__('latest item'), __('latest news'), __('recent items'), __('recent news'),  __('next items'), __('headlines')],
+
+  attributes: {
+    listHasDate: {
+      type: "boolean",
+      default: true
+    },
+    listHasTime: {
+      type: "boolean",
+      default: false
+    },
+    listLength: {
+      type: "numeric",
+      default: 3
+    },
+    listExpiry: {
+      type: "numeric",
+      default: 0
+    },
+    listEmptyText: {
+      type: "string",
+      default: "No items to display."
+    },
+    listItemType: {
+      type: "string",
+      default: "post"
+    },
+    listSort: {
+      type: "string",
+      default: "date"
+    },
+    listDatePrefix: {
+      type: "string",
+      default: "Updated"
+    },
+    listImage: {
+      type: "boolean",
+      default: false
+    },
+    listDefaultImage: {
+      type: "string",
+      default: ""
+    },
+    pastFuture: {
+      type: "string",
+      default: "past"
+    }
+
+  },
+
+  edit: props => {
+    const {
+      setAttributes,
+      attributes: {
+        listLength,
+        listExpiry,
+        listEmptyText,
+        listHasDate,
+        listHasTime,
+        listItemType,
+        listSort,
+        listDatePrefix,
+        listImage,
+        listDefaultImage,
+        pastFuture
+      },
+      className
+    } = props
+
+    const {
+      allPostTypes,
+    } = useSelect(
+      ( select ) => {
+          const { getPostTypes } = select(
+          coreStore
+        );
+        const allPostTypesList = getPostTypes(
+          { per_page: -1 }
+        );
+        return {
+          allPostTypes: allPostTypesList
+        };
+      }
+    );
+
+    let itemTypes = [{
+      label: "-",
+      value: ""
+    }]
+    let itemTypesFinishedParsing = false;
+
+    let fieldOptions = [
+      {label: "Publish date", value: "date"},
+      {label: "Modified date", value: "modified"},
+    ]
```

Comment:
Should modified date be here considering we dont offer that sort anywhere else nor do we display it anywhere?

Reply:
Removed (for now)